enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,291 @@
+another command to produce this output and avoid installing `oc`/`kubectl` is
+not the best, since we would be adding runtime dependencies between a system
+component and MicroShift.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```bash
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+1. CEE having a different tool.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
+
+If it is the former then we need to ensure newer MicroShift versions do not
+introduce incompatible changes with the reporting capabilities of the `sos`
+plugin. This may also be seen from a different perspective: the plugin should
+make no assumptions about names or locations of components whenever possible.
+In other words, the plugin must be written in the most generic way that still
+allows retrieval of relevant information.
+
+Still, if incompatible changes are introduced then we need to upgrade `sos` as
+well.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar as OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
+is an urgent need a hotfix can be made between releases.
+
+#### Extra disk space
+Adding a plugin to `sos` and not relying on MicroShift binary means it needs
+the `oc` or `kubectl` binary installed. This package is available in the RHEL
+repos, but it comes at an extra cost of 125MB once installed.
+
+If we compare it against the minimum disk size required (which is 8GB at the
+time of this writing), it represents an extra 1.5%.
+
+The greenboot feature for MicroShift 4.13 requires installing `oc` as well, so
+the first enhancement to be implemented will introduce it for the other one.
+
+#### CEE having a different tool
+CEE might have concerns over using a different tool than `must-gather`. `sos`
+reports are able to extract a full system report, as opposed to `must-gather`
+which is only suitable for OpenShift clusters.
```

Comment:
I like the idea about having the reporting evolve at the same time than the product but this requires thinking about the output format.
This would be only for cluster resources retrieval, right? As it needs `microshift` to be running and that might not always be the case.
It is easy to end up changing the plugin every time we change something in the `microshift report` command.
Seems to go a bit against the `sos` philosophy, judging by what I see in other plugins.

Reply:
I think you've captured the reason not to implement the logic in sos already. 

I think the suggestion to add a must-gather command is similar to adding the inspect command, right?

Reply:
It is quite the same, like embedding must-gather inside microshift. I think we can resolve this.